### PR TITLE
Elixir Ockam TCP Transport Initial Implementation

### PR DESCRIPTION
**Pull request by mjc**
This adds the beginnings of a TCP transport to Ockam's Elixir codebase.
Additionally this switches to BARE wire encoding.

Currently every outgoing message opens a new TCP connection.
A follow-up must attach outgoing connections to `:ranch` and also implement `Ockam.Worker` on the ranch handler, so that return messages go to the correct location.

### CODE REVIEW

File: implementations/elixir/ockam/ockam/mix.exs [hunk 1]
```diff
@@ -72,6 +74,7 @@ defmodule Ockam.MixProject do
 
   defp aliases do
     [
+      credo: "credo --strict",
```
Comment (mrinalwadhwa) [inline implementations/elixir/ockam/ockam/mix.exs:77]: there's another alias for this below `lint.credo`

Comment (mjc) [reply implementations/elixir/ockam/ockam/mix.exs:77]: Sure, I noticed that. However, I was running `mix credo` automatically in VSCode and did not look at the alias list.  It seems like switching the default to what we expect would keep others from falling into the same trap.  I'd argue that we should do that on all of our elixir packages, too - but didn't think about it til just now.

Comment (mrinalwadhwa) [reply implementations/elixir/ockam/ockam/mix.exs:77]: cool, makes sense 👍

**Comment by mrinalwadhwa** [comment]
974551305c918836b60ce68de46ec607ce9410e8 - could you change this commit message to describe the changes in the commit.

**Comment by mrinalwadhwa** [comment]
Awesome, looks ready to merge, one last thing ... the commit message got left mid sentence `test(elixir): add encoding/decoding tests for`

**Pull request merged**